WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift

```diff
@@ -38,30 +38,38 @@ final class ShippingLabelAddressFormViewController: UIViewController {
             guard let self = self else { return }
             ServiceLocator.analytics.track(.shippingLabelEditAddressContactCustomerButtonTapped)
 
-            let actionSheet = UIAlertController(title: nil, message: Localization.contactActionLabel, preferredStyle: .actionSheet)
-            actionSheet.view.tintColor = .text
-
-            actionSheet.addCancelActionWithTitle(Localization.contactActionCancel)
-            if let email = email, email.isNotEmpty {
-                actionSheet.addDefaultActionWithTitle(Localization.contactActionEmail) { _ in
-                    self.sendEmail(to: email)
-                }
-            }
-            if let phoneNumber = phone, phoneNumber.isNotEmpty {
-                actionSheet.addDefaultActionWithTitle(Localization.contactActionCall) { _ in
-                    if PhoneHelper.callPhoneNumber(phone: phoneNumber) == false {
-                        self.displayErrorNotice(title: Localization.phoneNumberErrorNotice)
+            if PhoneHelper.canCallPhoneNumber(phone: "123456789") == false
+                && MFMailComposeViewController.canSendMail() == false
+                && MFMessageComposeViewController.canSendText() == false {
```

Comment:
nit: This works, but just to offer another way to check that these values are `false`:

```
            if !PhoneHelper.canCallPhoneNumber(phone: "123456789")
                && !MFMailComposeViewController.canSendMail()
                && !MFMessageComposeViewController.canSendText() {
```

You could also use a `guard` statement here, rather than an `if`/`else`, if you want to avoid nesting the expected behavior in an `else` statement. That's again about style more than anything, but I find that it can help to read the code when the desired or expected behavior is less nested.

Don't feel like you have to make any changes based on nits like these! I'm just offering them as alternatives to consider.

Reply:
Thank you so much for these tips! I appreciate all the suggestions and nits that help me improve my code's readability and coding style! I haven't gone through the tutorial about using `guard`, but I'll look it up!